server/src/main/java/org/opensearch/cluster/routing/IndexShardRoutingTable.java

```diff
@@ -686,6 +686,23 @@ public ShardIterator replicaFirstActiveInitializingShardsIt() {
         return new PlainShardIterator(shardId, ordered);
     }
 
+    /**
+     * Returns an iterator on replica shards.
+     */
+    public ShardIterator searchReplicaActiveInitializingShardIt() {
```

Comment:
I think you could pull out an internal method that takes a `Predicate<ShardReplica> filter` as an argument and share the implementation with the other version of this method that doesn't filter out non-search replicas.

Reply:
added